src/query_graph/path_tree.rs

```diff
@@ -111,13 +111,59 @@ impl OpPathTree {
         }
         Ok(true)
     }
+
+    fn fmt_internal(
+        &self,
+        f: &mut Formatter<'_>,
+        indent: &String,
+        include_conditions: bool,
+    ) -> std::fmt::Result {
+        if self.is_leaf() {
+            return writeln!(f, "{}", self.vertex());
```

Comment:
I think in this case we should not add a newline, as it's a one-line path string? The JS code doesn't add a newline here and it makes sense for the caller to decide if it should be followed by a newline or not.
```suggestion
            return write!(f, "{}", self.vertex());
```